.github/workflows/ci.yml

```diff
@@ -103,9 +103,9 @@ jobs:
           CI_FOR_PR: true
 
       - name: Save coverage results
-        uses: actions/upload-artifact@0b7f8abb1508181956e8e162db84b466c27e18ce # tag v3.1.2
+        uses: actions/upload-artifact@0b2256b8c012f0828dc542b3febcab082c67f72b # tag v4.3.4
         with:
-          name: coverage-report-unit-tests
+          name: coverage-report-unit-tests-${{ matrix.ruby-version }}-${{ env.rails }}
```

Comment:
Upgrading from v3 to v4 requires artifacts to have different names as they are immutable, unlike in v3. 

Migration guide: https://github.com/actions/upload-artifact/blob/main/docs/MIGRATION.md

Reply:
Will this change impact anything such as the code coverage stitcher or Stoat? If so, we should document those with new GitHub Issues and weigh whether or not we should upgrade this one or hold off until the child issues are addressed.

Reply:
@fallwith I'm not seeing any issues with code coverage being stitched (example run w/code coverage at the bottom: https://github.com/newrelic/newrelic-ruby-agent/actions/runs/10167351518).

I'm not however sure how Stoat might be impacted (I'm not sure how stoat would use this code). Is this something you know off the top of your head?

Reply:
I was coming at this from the perspective of all artifact usage involving two steps: an upload step, and a download step; with the download step being optional in some cases where we might want to store results for a potential manual review that never ends up being needed.

The line I was commenting on is impacting the upload step by changing the name of the upload. So that made me wonder if the corresponding download step(s) would be impacted and need to be updated to be aware of the new name.

It looks like this repo's own downloading logic is tolerant to the name change because this `ci.yml` file's `simplecov` job seems to just grab ALL artifacts from the workflow without caring about the names. The actual stitcher at `lib/tasks/coverage_report.rake` seems to make use of splats (`*`) to not care about the name change either as long as the names still start with `coverage_`. So we're good there with this repo.

Stoat has fully three separate subcommands that all make use of the `fetch_artifact` helper, so I wondered if it might be affected as a downloader of artifacts uploaded by this repo's actions.

It seems like for this `ci.yml` file, it focuses only on `coverage-report-combined` which this PR does not rename, so we are ok.